store/store.go

```diff
@@ -429,7 +460,8 @@ func (s *Store[H]) Append(ctx context.Context, headers ...H) error {
 func (s *Store[H]) flushLoop() {
 	defer close(s.writesDn)
 	ctx := context.Background()
```

Comment:
side note - we never cancel this ctx if we exit, should we put a cancellation in defer ?

Reply:
We can and even should if any internal calls spins up a backrground routine waiting on context, but as its not happening anywhere yet, it fine to keep as is